### PR TITLE
STORM_B add printer page to admin

### PR DESCRIPTION
**Pull request by AndriiBuhalo**
Task: Create new colum Printer in admin panel.
image

Problem in this task:
Can't create a lot of child in parent for printer relations.
image
image
After save a have only last child(

### CODE REVIEW

**Comment by AndriiBuhalo** [comment]
Don't work update printer. I don't understand why i can't stand up in method update(

File: app/views/dredd/products/_product_option_color_values_form.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="nested-form-wrapper" data-new-record="<%= f.object.new_record? %>">
+z<div class="nested-form-wrapper" data-new-record="<%= f.object.new_record? %>">
```
Comment (Shabaldas) [inline app/views/dredd/products/_product_option_color_values_form.erb:1]: z???
smells russian swastika 😄

File: app/views/dredd/products/_form.html.erb [hunk 2]
```diff
@@ -40,8 +40,6 @@
       <div data-target="toggler.hideme" id="stuff_to_hide">
         <%= form.fields_for :primary_product_option do |primary_option| %>
           <%= primary_option.label t("dredd.products.primary") %>
-          <%= primary_option.check_box :primary, { checked: true }, true, false  %>
```
Comment (Shabaldas) [inline app/views/dredd/products/_form.html.erb:43]: but why???
DON'T TOUCH PRODUCT

please.

File: app/views/dredd/products/_form.html.erb [hunk 3]
```diff
@@ -40,8 +40,6 @@
...
-          <%= primary_option.hidden_field :option_id, value: Option.last.id %>
```
Comment (Shabaldas) [inline app/views/dredd/products/_form.html.erb:44]: leave like before

File: app/views/dredd/products/_form.html.erb [hunk 4]
```diff
@@ -64,8 +62,6 @@
       <input data-action="input->toggler#toggle" data-target="toggler.input" type="checkbox">
       <div data-target="toggler.hideme" id="stuff_to_hide"  >
         <%= form.fields_for :secondary_product_option do |secondary_option| %>
-          <%= secondary_option.label t("dredd.products.secondary") %>
```
Comment (Shabaldas) [inline app/views/dredd/products/_form.html.erb:67]: leave like before

File: app/views/dredd/products/_form.html.erb [hunk 5]
```diff
@@ -64,8 +62,6 @@
...
-          <%= secondary_option.hidden_field :option_id, value: Option.first.id %>
```
Comment (Shabaldas) [inline app/views/dredd/products/_form.html.erb:68]: leave like before

File: config/locales/simple_form.en.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+en:
```
Comment (Shabaldas) [inline config/locales/simple_form.en.yml:1]: remove all this file

File: lib/templates/erb/scaffold/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<%# frozen_string_literal: true %>
```
Comment (Shabaldas) [inline lib/templates/erb/scaffold/_form.html.erb:1]: remove all this file

File: app/controllers/dredd/printer_maintenances_controller.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
```
Comment (Shabaldas) [inline app/controllers/dredd/printer_maintenances_controller.rb:1]: so we discuss to use only one controller, why do you need this one?

File: app/controllers/dredd/printers_controller.rb [hunk 9]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Dredd
+  class PrintersController < BaseController
+    before_action :set_printer, only: [:edit, :update, :destroy]
+
+    def index
+      @pagy, @printers = pagy(Printer.all, items: 10)
+      @printer = Printer.first
```
Comment (Shabaldas) [inline app/controllers/dredd/printers_controller.rb:9]: but why do you need this variable?

File: app/controllers/dredd/printers_controller.rb [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    end
+
+    def new
+      @printer = Printer.new
+      @printer_maintenance = []
+      # @printer.printer_maintenance.build
+      6.times { @printer.printer_maintenance.build }
```
Comment (Shabaldas) [inline app/controllers/dredd/printers_controller.rb:16]: what is this?
😄

File: app/controllers/dredd/printers_controller.rb [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    before_action :set_printer, only: [:edit, :update, :destroy]
+
+    def index
+      @pagy, @printers = pagy(Printer.all, items: 10)
+      @printer = Printer.first
+    end
+
+    def new
+      @printer = Printer.new
+      @printer_maintenance = []
```
Comment (Shabaldas) [inline app/controllers/dredd/printers_controller.rb:14]: you don't need it

File: app/controllers/dredd/printers_controller.rb [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    private
+
+    def product_params
+      params.require(:printer).permit(:printer_code, :firm, :model, :printing_technology, :state,
+                                      :type_mechanic, :table_size, :price_for_printer, :bought,
+                                      :comment, :by_for_upgrade,
+                                      printer_maintenance_attributes: [:id, :problem, :problem_find, :time_for_fix, :_destroy])
+    end
+
+    def prepare_params(product_params)
```
Comment (Shabaldas) [inline app/controllers/dredd/printers_controller.rb:54]: I don't understand why do you need it.
but leave for now.

**Review by Shabaldas — CHANGES_REQUESTED** [review]
Please change this issues, and after that we come back.
And please update your description 🥺 
Where is pictures?
Don't forget that this is not simple PR, this is a document 🙏

File: app/views/dredd/products/_product_option_color_values_form.erb (hunk 1 above)
Comment (AndriiBuhalo) [reply app/views/dredd/products/_product_option_color_values_form.erb]: Fix

File: app/views/dredd/products/_form.html.erb (hunk 2 above)
Comment (AndriiBuhalo) [reply app/views/dredd/products/_form.html.erb]: OK

File: app/views/dredd/products/_form.html.erb (hunk 4 above)
Comment (AndriiBuhalo) [reply app/views/dredd/products/_form.html.erb]: Fix

File: app/controllers/dredd/printers_controller.rb (hunk 9 above)
Comment (AndriiBuhalo) [reply app/controllers/dredd/printers_controller.rb]: Fix

**Review by Shabaldas — APPROVED** [review]

**Pull request merged**